angular/whereami/src/app/views/create-game/handpicked-game-form/handpicked-game-form.component.ts

```diff
@@ -41,4 +62,42 @@ export class HandpickedGameFormComponent implements OnInit {
   public cancel() {
     this.cancelled.emit();
   }
+
+  openModal() {
+    this.modalService.open(this.importFileTooltip);
+  }
+
+  handleImportFile(e: any) {
+    this.statusText = "";
+    const file: File = e.target.files[0];
+    if (file) {
+      this.selectedFile = file.name;
+      const reader = new FileReader();
+      reader.readAsText(file, "UTF-8");
+      reader.onload = (evt: ProgressEvent<FileReader>) => {
+        try {
+          var fileContent = evt.target!.result;
+          var obj = JSON.parse(fileContent!.toString());
+          if (this.model.name.trim() == "") {
+            this.model.name = obj.map_name;
+          }
+          var locations = obj.locations;
+          for (var locId in locations) {
+            var location = locations[locId];
+            if (location.hasOwnProperty('name') && location.name.trim() != "") {
+              this.handPickedManager!.addMarker(location.latLng, location.name);
+            } else {
+              const importRadius = 50; // for imports there should be a street view panorama in close distance
+              this.handPickedManager!.addMarkerFromPanorama(location.latLng, importRadius); // retrieve a panorama for the location to get a name
+            }
+          }
+        } catch (e) {
+          this.statusText = "Error reading file: json syntax invalid";
+          console.log(e);
+        }
+      };
+    } else {
+      this.statusText = "no file selected"
+    }
+  }
```

Comment:
Please find a way to decuple file+parsing logic from component. 
Please use/create an model/interface for the file's json structure. 

I would suggest an own service within the create game feature folder: 
form-component -> importservice.parse(file) => result: some intermediate object
form-component -> handPickedManager(intermediate object)

PS: Angular HttpClient can be used to parse a Json file.